Include a java too new path in android-java-gradle-migration (#11708)
flutter/website: src/content/release/breaking-changes/android-java-gradle-migration-guide.md

   $ ./gradlew wrapper --gradle-version=7.6.1
   ```

> ## You didnt update android studio and still have a java error

Reviewer: nit: missing apostrophe in "didn't"
r/android studio/Android Studio
r/java/Java

Author: ```suggestion
## You didn't update Android Studio and still have a Java error
```

The error appears similar to `Unsupported class file major version 65`. 

> This is an indication that your java version is newer than the version of

Reviewer: r/java/Java everywhere

Author: ```suggestion
This is an indication that your Java version is newer than the version of
```

gradle you are running can handle. There is a non obvious set of dependencies
surrounding AGP, java, and gradle. 

> ### Solution 1: Use Android Studio 

Reviewer 2: ```suggestion
### Solution 1: Android Studio 
```

The easiest way to resolve this issue is to use Android Studio AGP upgrade assistant. 

> To use select your top level build.gradle file in Android Studio then select 

Reviewer 2: ```suggestion
To use select your top-level `build.gradle` file in Android Studio then select 
```

Tools -> AGP Upgrade Assistant. 

### Solution 2: Command line

> Run `flutter analyze --suggestion` to see if your AGP, Java and gradle versions are compatible. 

Reviewer 2: ```suggestion
Run `flutter analyze --suggestion` to see if your AGP, Java, and gradle versions are compatible. 
```

If Gradle needs to be updated you can update it with `./gradlew wrapper --gradle-version=SOMEGRADLEVERSION`

> where SOMEGRADLEVERSION is the version suggested by flutter analyze or newer. 

Reviewer 2: ```suggestion
where SOMEGRADLEVERSION is the version (you can use a newer version)
suggested by `flutter analyze`. 
```

> To find the java version being used run `flutter doctor`
> On a mac you can find the versions of java the OS knows about with `/usr/libexec/java_home -V` 
> To set the version of java that all flutter projects use run `flutter config --jdk-dir=SOMEJAVAPATH`
> where SOMEJAVAPATH is a path to a java version like `/opt/homebrew/Cellar/openjdk@17/17.0.13/libexec/openjdk.jdk/Contents/Home`

Reviewer 2: ```suggestion
To find the Java version being used run `flutter doctor`.
On a mac, you can find the Java versions that the OS knows about with `/usr/libexec/java_home -V`.
To set the version of Java that all flutter projects use run `flutter config --jdk-dir=SOMEJAVAPATH`
where SOMEJAVAPATH is a path to a Java version like `/opt/homebrew/Cellar/openjdk@17/17.0.13/libexec/openjdk.jdk/Contents/Home`
```

   $ ./gradlew wrapper --gradle-version=7.6.1
   ```

## You didnt update android studio and still have a java error
The error appears similar to `Unsupported class file major version 65`. 
This is an indication that your java version is newer than the version of
gradle you are running can handle. There is a non obvious set of dependencies

> surrounding AGP, java, and gradle. 

Author: ```suggestion
surrounding AGP, Java, and Gradle. 
```

### Solution 1: Android Studio 
The easiest way to resolve this issue is to use Android Studio AGP upgrade assistant. 
To use select your top-level `build.gradle` file in Android Studio then select 
Tools -> AGP Upgrade Assistant. 

### Solution 2: Command line

> Run `flutter analyze --suggestion` to see if your AGP, Java, and gradle versions are compatible. 

Author: ```suggestion
Run `flutter analyze --suggestion` to see if your AGP, Java, and Gradle versions are compatible. 
```
